### PR TITLE
[Refactor](agg-func) Remove some nonsense template interlayers

### PR DESCRIPTION
**Pull request by zclllyybb**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

In this pr, we significantly improved code readability.

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Review by zhangstar333 — DISMISSED** [review]

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/aggregate_functions/aggregate_function_covar.h [hunk 1]
```diff
@@ -122,23 +105,30 @@ struct BaseData {
...
+    double sum_x {};
+    double sum_y {};
+    double sum_xy {};
+    int64_t count {};
 };
 
-template <typename T, typename Data>
-struct PopData : Data {
+template <typename T>
+struct PopData : BaseData<T> {
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_covar.h:115]: why call name "covar" ?

File: be/src/vec/aggregate_functions/aggregate_function_covar.h [hunk 2]
```diff
@@ -122,23 +105,30 @@ struct BaseData {
...
+    static constexpr bool is_pop = true;
+    static const char* name() { return "covar"; }
+    static DataTypePtr get_return_type() { return std::make_shared<DataTypeNumber<Float64>>(); }
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_covar.h:118]: should impl in `BaseData<T>` always Float64

File: be/src/vec/aggregate_functions/aggregate_function_stddev.h [hunk 3]
```diff
@@ -210,14 +195,7 @@ class AggregateFunctionSampVariance
...
         } else {
-            if constexpr (is_nullable) { //this if check could remove with old function
-                const auto* nullable_column = check_and_get_column<ColumnNullable>(columns[0]);
-                if (!nullable_column->is_null_at(row_num)) {
-                    this->data(place).add(&nullable_column->get_nested_column(), row_num);
-                }
-            } else {
-                this->data(place).add(columns[0], row_num);
-            }
+            this->data(place).add(columns[0], row_num);
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_stddev.h:198]: so why we need if?

**Review by HappenLee — CHANGES_REQUESTED** [review]
need detail code review @zhangstar333

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/aggregate_functions/aggregate_function_covar.h (hunk 1 above)
Comment (zclllyybb) [reply be/src/vec/aggregate_functions/aggregate_function_covar.h:116]: covar = covar_pop != covar_samp

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/aggregate_functions/aggregate_function_stddev.h [hunk 4]
```diff
@@ -191,14 +165,25 @@ struct SampData : Data {
...
+    static const char* name() { return "variance"; }
+};
+struct VarianceSampName {
+    static const char* name() { return "variance_samp"; }
+};
+struct StddevSampName {
+    static const char* name() { return "stddev_samp"; }
+};
+
+template <bool is_pop, typename Data>
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_stddev.h:181]: so what is_pop use? seems unless?

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**